src/components/SearchBar.js

```diff
@@ -223,6 +357,44 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
     }
   }
 
+  // saves the current search from the url and feed it to a separate index zone
+  saveThisSearch(searcherState) {
+    const searcher = this.context.searcher;
+    const { savedSearchZone, savedSearchTable } = this.props;
+    const currentSearch = searcher.generateLocationQueryStringFromState(searcherState);
+    const username = AuthUtils.getLoggedInUserId();
+    const loggedDateTime = new Date().toISOString();
+    const id = username.concat(loggedDateTime);
+    const title = this.state.savedSearchTitle;
```

Comment:
Place this at the top of the function somewhere above `line 364`. Prefer destructuring:
`const { savedSearchTitle: title } = this.state;`